packages/hdx_zed/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use std::env;
+use std::fs;
+use zed_extension_api::{self as zed, Result};
+
+struct HdxExtension {
+	cached_binary_path: Option<String>,
+}
+
+impl HdxExtension {
+	fn language_server_binary_path(
+		&mut self,
+		language_server_id: &zed_extension_api::LanguageServerId,
+		worktree: &zed::Worktree,
+	) -> Result<String> {
+		if let Some(path) = worktree.which("hdx") {
+			return Ok(path);
+		}
+
+		if let Some(path) = &self.cached_binary_path.clone().or_else(|| env::var("HDX_SERVER_PATH").ok()) {
+			if fs::metadata(path).map_or(false, |stat| stat.is_file()) {
+				return Ok(path.to_string());
+			}
+		}
+
+		zed::set_language_server_installation_status(
+			language_server_id,
+			&zed::LanguageServerInstallationStatus::CheckingForUpdate,
+		);
+
+		let release = zed::github_release_by_tag_name("keithamus/hdx", "canary")?;
+
+		let (platform, arch) = zed::current_platform();
+		let asset_name = format!(
+			"binary-{platform}-{arch}",
+			platform = match platform {
+				zed::Os::Mac => "darwin",
+				zed::Os::Linux => "linux",
+				zed::Os::Windows => "win32",
+			},
+			arch = match arch {
+				zed::Architecture::Aarch64 => "arm64",
+				zed::Architecture::X8664 => "x64",
+				_ => return Err(format!("unsupported architecture: {arch:?}")),
+			},
+		);
+
+		let asset = release
+			.assets
+			.iter()
+			.find(|asset| asset.name == asset_name)
+			.ok_or_else(|| format!("no asset found matching {:?}", asset_name))?;
+
+		if !fs::metadata(&asset_name).map_or(false, |stat| stat.is_file()) {
+			zed::set_language_server_installation_status(
+				language_server_id,
+				&zed::LanguageServerInstallationStatus::Downloading,
+			);
+
+			zed::download_file(&asset.download_url, "bin", zed::DownloadedFileType::Zip)
```

Comment:
Once we have the release assets going, this line may need to change a tad. As in is it `bin/binary-platform-arch`??